src/bucket/BucketManagerImpl.cpp

```diff
@@ -894,20 +894,20 @@ BucketManagerImpl::addBatch(Application& app, uint32_t currLedger,
 #ifdef BUILD_TESTS
     if (mUseFakeTestValuesForNextClose)
     {
-        currLedgerProtocol = mFakeTestProtocolVersion;
+        header.ledgerVersion = mFakeTestProtocolVersion;
```

Comment:
mm, well, i get the desire to avoid copying as much as possible, but in this PR it's not about supporting a test case, but rather it's a safety choice. I purposely didn't want to pass around LedgerHeader references, because they can get invalidated by LedgerTxn and we'd be left with dangling references. Headers shouldn't take up kilobytes, right? It's just a few hashes and uints, and we're only doing this once a ledger. So I'd prefer to keep this function as-is.